lib/src/builders/soundboard.dart

```diff
@@ -0,0 +1,47 @@
+import 'package:nyxx/src/builders/builder.dart';
+import 'package:nyxx/src/builders/sentinels.dart';
+import 'package:nyxx/src/builders/sound.dart';
+import 'package:nyxx/src/models/emoji.dart';
+import 'package:nyxx/src/models/soundboard/soundboard.dart';
+
+Map<String, String> _makeEmojiMap(Emoji? emoji) => {
+      if (emoji case final TextEmoji emoji?) 'emoji_name': emoji.name,
+      if (emoji case final GuildEmoji emoji?) 'emoji_id': emoji.id.toString(),
+    };
+
+class SoundboardSoundBuilder extends CreateBuilder<SoundboardSound> {
+  String name;
+
+  SoundBuilder sound;
+
+  double? volume;
+
+  Emoji? emoji;
+
+  SoundboardSoundBuilder({required this.name, required this.sound, this.volume, this.emoji});
+
+  @override
+  Map<String, Object?> build() => {
+        'name': name,
+        'sound': sound.buildDataString(),
+        if (volume != null) 'volume': volume,
+        ..._makeEmojiMap(emoji),
+      };
+}
+
+class UpdateSoundBuilder extends UpdateBuilder<SoundboardSound> {
+  String name;
+
+  double volume;
+
+  Emoji emoji;
+
+  UpdateSoundBuilder({required this.name, this.volume = sentinelDouble, this.emoji = sentinelEmoji});
```

Comment:
Actually... Why does this need a sentinel if it can't be set to null?

Just make it nullable, and don't include it if it's null.

Reply:
It can be set to null, I forgot to include the nullable type